### PR TITLE
Fix OVERVIEW_LEVEL=NONE open option

### PR DESCRIPTION
**Pull request by rouault**
and also "{x}only". This however needs cooperation from drivers.
I've just updated the GTiff, JP2KAK, JPEG, OpenJPEG and VRT ones. Others
would need similar changes in their specific GetOverviewCount() / IRasterIO()
implementation.

Fixes https://github.com/OSGeo/gdal/pull/3181#issuecomment-822868290

### CODE REVIEW

File: autotest/gcore/overviewds.py [hunk 1]
```diff
@@ -105,6 +122,8 @@ def test_overviewds_2():
     assert ds.GetMetadataItem('RPC', 'FOO') is None
     ds = None
 
+    gdal.GetDriverByName('GTiff').Delete('tmp/byte.tif')
```
Comment (sgillies) [inline autotest/gcore/overviewds.py:125]: @rouault if we use pytest's `tmp_path` fixture, there's no need for manual cleanup of the temp files.

File: autotest/gcore/overviewds.py [hunk 2]
```diff
@@ -48,31 +48,46 @@ def test_overviewds_1():
 ###############################################################################
 # Nominal cases
 
-
-def test_overviewds_2():
+@pytest.mark.parametrize("externalOverviews", [True, False])
+def test_overviewds_2(externalOverviews):
```
Comment (sgillies) [inline autotest/gcore/overviewds.py:52]:
```suggestion
def test_overviewds_2(tmp_path, externalOverviews):
```

File: autotest/gcore/overviewds.py [hunk 3]
```diff
@@ -48,31 +48,46 @@ def test_overviewds_1():
...
 
     shutil.copy('data/byte.tif', 'tmp')
```
Comment (sgillies) [inline autotest/gcore/overviewds.py:54]:
```suggestion
    shutil.copy('data/byte.tif', tmp_path)
```

File: autotest/gcore/overviewds.py [hunk 4]
```diff
@@ -48,31 +48,46 @@ def test_overviewds_1():
...
-    ds = gdal.Open('tmp/byte.tif')
-    ds.BuildOverviews('NEAR', overviewlist=[2, 4])
+    if externalOverviews:
+        ds = gdal.Open('tmp/byte.tif')
```
Comment (sgillies) [inline autotest/gcore/overviewds.py:56]:
```suggestion
        ds = gdal.Open(str(tmp_path.joinpath('byte.tif')))
```

Comment (sgillies) [reply autotest/gcore/overviewds.py:56]: &c.

**Comment by sgillies** [comment]
@rouault do I understand correctly that we'll release 3.3.0 with possible bugs in other drivers? That's okay with me.

**Comment by rouault** [comment]
> do I understand correctly that we'll release 3.3.0 with possible bugs in other drivers?

yes, that might not work properly in a few other drivers that have specialized behavior regarding use of overviews.

**Pull request merged**